src/frameworks/next/index.ts

```diff
@@ -367,41 +393,75 @@ export async function ɵcodegenPublicDirectory(sourceDir: string, destDir: strin
 
   await Promise.all(
     Object.entries(routesToCopy).map(async ([path, route]) => {
-      if (
-        route.initialRevalidateSeconds ||
-        pathsUsingsFeaturesNotSupportedByHosting.some((it) => path.match(it))
-      ) {
+      if (route.initialRevalidateSeconds) {
+        if (process.env.DEBUG) console.log(`skipping ${path} due to revalidate`);
+        return;
+      }
+      if (pathsUsingsFeaturesNotSupportedByHosting.some((it) => path.match(it))) {
+        if (process.env.DEBUG)
+          console.log(
+            `skipping ${path} due to it matching an unsupported rewrite/redirect/header or middlware`
+          );
         return;
       }
-
       const appPathRoute =
         route.srcRoute && appPathRoutesEntries.find(([, it]) => it === route.srcRoute)?.[0];
       const contentDist = join(sourceDir, distDir, "server", appPathRoute ? "app" : "pages");
 
-      const parts = path.split("/").filter((it) => !!it);
-      const partsOrIndex = parts.length > 0 ? parts : ["index"];
+      const sourceParts = path.split("/").filter((it) => !!it);
+      const locale = i18n?.locales.includes(sourceParts[0]) ? sourceParts[0] : undefined;
+      const includeOnThisDomain =
+        !locale ||
+        !matchingI18nDomain ||
+        matchingI18nDomain.defaultLocale === locale ||
+        !matchingI18nDomain.locales ||
+        matchingI18nDomain.locales.includes(locale);
+
+      if (!includeOnThisDomain) {
+        if (process.env.DEBUG)
+          console.log(`skipping ${path} since it is for a locale not deployed on this domain`);
+        return;
+      }
 
-      let sourcePath = join(contentDist, ...partsOrIndex);
-      let destPath = join(destDir, ...partsOrIndex);
+      const sourcePartsOrIndex = sourceParts.length > 0 ? sourceParts : ["index"];
+      const destParts = sourceParts.slice(locale ? 1 : 0);
+      const destPartsOrIndex = destParts.length > 0 ? destParts : ["index"];
+      const isDefaultLocale = !locale || (matchingI18nDomain || i18n)?.defaultLocale === locale;
+
+      let sourcePath = join(contentDist, ...sourcePartsOrIndex);
+      let localizedDestPath =
+        !singleLocaleDomain &&
+        locale &&
+        join(destDir, I18N_ROOT, locale, basePath, ...destPartsOrIndex);
+      let defaultDestPath = isDefaultLocale && join(destDir, basePath, ...destPartsOrIndex);
```

Comment:
is this something we could turn into a util? Looks harder for me to understand now that we're handling i18n.. if not maybe some comments could help

Reply:
yeah, we should write utils for turning the path and route into a filesystem path. let's take as a follow on